pkg/reconciler/pipelinerun/resources/resultrefresolution.go

```diff
@@ -109,64 +109,73 @@ func removeDup(refs ResolvedResultRefs) ResolvedResultRefs {
 	return deduped
 }
 
-// convertToResultRefs walks a PipelineTask looking for result references. If any are
-// found they are resolved to a value by searching pipelineRunState. The list of resolved
-// references are returned. If an error is encountered due to an invalid result reference
-// then a nil list and error is returned instead.
 func convertToResultRefs(pipelineRunState PipelineRunState, target *ResolvedPipelineTask) (ResolvedResultRefs, string, error) {
 	var resolvedResultRefs ResolvedResultRefs
-	for _, ref := range v1.PipelineTaskResultRefs(target.PipelineTask) {
-		resolved, pt, err := resolveResultRef(pipelineRunState, ref)
-		if err != nil {
-			return nil, pt, err
+	for _, resultRef := range v1.PipelineTaskResultRefs(target.PipelineTask) {
+		referencedPipelineTask := pipelineRunState.ToMap()[resultRef.PipelineTask]
+		if referencedPipelineTask == nil {
+			return nil, resultRef.PipelineTask, fmt.Errorf("could not find task %q referenced by result", resultRef.PipelineTask)
+		}
+		if !referencedPipelineTask.isSuccessful() && !referencedPipelineTask.isFailure() {
+			return nil, resultRef.PipelineTask, fmt.Errorf("task %q referenced by result was not finished", referencedPipelineTask.PipelineTask.Name)
+		}
+		switch {
```

Comment:
Looks like the switch statement was missing the condition expression? Shall we move the first case (IsCustomTask) out of the switch statement and move the switch condition onto the len of referencedPipelineTask.TaskRuns might be clearer here?

Reply:
Cleaned this up using if/else statements instead

Reply:
Oh my previous point was that the original if else statement might be clearer with your current refactor (I've observed you tried to refactor resolveCustomResultRef out). Also given that the switch statement was not actually switching on any conditon.

Don't want to block this PR but might be helpful for future PRs to have refactor PRs separated from the actual implementation PR for more readability.